src/main/scala/org/scanet/optimizers/AMSGrad.scala

```diff
@@ -0,0 +1,31 @@
+package org.scanet.optimizers
+
+import org.scanet.core._
+import org.scanet.syntax._
+
+case class AMSGrad(rate: Output[Float], beta1: Output[Float], beta2: Output[Float], epsilon: Output[Float]) extends Algorithm {
+
+  override def initMeta(shape: Shape): Tensor[Float] = {
+    val momentum = Tensor.zeros[Float](shape).const
+    val velocity = Tensor.zeros[Float](shape).const
+    val correctedVelocity = Tensor.zeros[Float](shape).const
+    val suffix = Tensor.zeros[Float](shape).const
+    ((momentum zip velocity) zip (correctedVelocity zip suffix)).eval
+  }
+
+  override def delta(grad: Output[Float], meta: Output[Float], iter: Output[Int]): Delta = {
+    val (args1, args2) = meta.unzip
+    val (prevMomentum, prevVelocity) = args1.unzip
+    val (prevCorrectedVelocity, suffix) = args2.unzip
```

Comment:
Maybe we can name `variance` variable as:
- `vPrev` for previous `variance`
- `vCurrent` for current `variance` 
- `v` result `variance` which is output of `max` functinon